### DISCUSSION TITLE
Unable to Send setInterval Command via MQTT in zwave-js API

### DISCUSSION THREAD

**Discussion by Sergey8100** [zwave-js/zwave-js-ui#3987] [Q&A]
When attempting to send the setInterval command to a Z-Wave device using the zwave-js MQTT API, I consistently receive errors related to argument formatting. Despite following the documented examples and API requirements, the command fails to execute, and the response indicates issues with interpreting the wakeUpInterval and controllerNodeId arguments.

MQTT Payload:
```
{
    "args": [
        {
            "nodeId": 11,
            "commandClass": 132,
            "endpoint": 0
        },
        "setInterval",
        [300, 1]
    ]
}
```

The API responds with errors such as:

- wakeUpInterval is not a number (ZW0322)
- if args is given, it must be an array
- Cannot read properties of undefined (reading 'nodeId')

`{"success":false,"message":"Cannot read properties of undefined (reading 'nodeId')","args":[],"origin":"{\n\t\"args\": [\n\t\t{\n        \"nodeId\": 11\n        \"endpoint\": 0\n        \"commandClass\": 132\n\t\t},\n\t\t\"setInterval\",\n\t\t[100]\n\t]\n}"}`

Could you help me. I think I've tried all the options already.

ps Command with SoftReset is working for me.

**Comment by robertsLando** [comment]
Could you tell me what mqtt topic you are using?

**Reply by Sergey8100** [reply]
yes, of course, here it is: zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/api/sendCommand/set

zwave-js-ui - name of the MQTT topic from settings

**Reply by robertsLando** [reply]
I have a feel that's something related to the formatting you are using in the JSON message as I'm not able to reproduce the issue. Could you try to send that json using mqtt explorer for example?

**Reply by Sergey8100** [reply]
It really worked. It is not clear why Mqtt Publish from Home Assistant does not work.

However, the data does not change on the device, although the API says that the command was successful.

**Reply by robertsLando** [reply]
@Sergey8100 try to call refreshCCValues api with cc 132

**Reply by Sergey8100** [reply]
Thanks it's working

**Reply by robertsLando** [reply]
Some devices after a value is updated don't emit an update to get the feedback. BTW if there is a valued associated to that value I suggest to use `/set` against the valueId instead of calling `sendCommand` api
